### PR TITLE
Fix issue 366, Problem when shaaring a link in Reader View of Firefox.

### PR DESCRIPTION
**Pull request by StyMaar**
A back-end fix for #366

### CODE REVIEW

**Comment by StyMaar** [comment]
Oh, I didn't even realized that this part of the code was covered with tests !
I'm gonna fix this asap.

File: application/Url.php [hunk 1]
```diff
@@ -113,6 +113,19 @@ public function __construct($url)
         }
     }
 
+
+    private function removeFirefoxAboutReader($input){
+      $output_array = [];
```
Comment (ArthurHoaro) [inline application/Url.php]: Please use `array()` for PHP 5.3 compatibility.

File: application/Url.php [hunk 2]
```diff
@@ -113,6 +113,19 @@ public function __construct($url)
         }
     }
 
+
+    private function removeFirefoxAboutReader($input){
```
Comment (ArthurHoaro) [inline application/Url.php]: I think you should either use the `$parts` attribute in your method, or create a function outside the class.

File: application/Url.php [hunk 3]
```diff
@@ -113,6 +113,19 @@ public function __construct($url)
...
+      $output_array = [];
+      preg_match("%^about://reader\?url=(.*)%", $input, $output_array);
```
Comment (ArthurHoaro) [inline application/Url.php]: Why are you using your regex twice?

**Comment by ArthurHoaro** [comment]
Thank you for your work! 

Tests are failing because of 5.3 compatibility, no big deal.

Also could you add Doxygen-like documentation to your method, and try to follow [our coding style](https://github.com/shaarli/Shaarli/issues/95#issuecomment-120890850).

File: application/Url.php (hunk 2 above)
Comment (virtualtam) [inline application/Url.php]: this method should be _static_ and called in the _constructor_, before exploding a URL's parts:
```php
public function __construct($url)
{
    $url = self::cleanupFirefoxReader($url);
    $this->parts = parse_url($url);
    ...
}
```

File: application/Url.php [hunk 4]
```diff
@@ -175,7 +188,8 @@ public function cleanup()
     {
         $this->cleanupQuery();
         $this->cleanupFragment();
-        return $this->toString();
+        $url = $this->toString();
```
Comment (virtualtam) [inline application/Url.php]: this method should be left untouched (see comment on constructor)

**Pull request closed without merging**